sermant-plugins/sermant-mq-grayscale/mq-grayscale-rocketmq-plugin/src/main/java/io/sermant/mq/grayscale/rocketmq/interceptor/RocketMqSchedulerRebuildSubscriptionInterceptor.java

```diff
@@ -35,44 +39,81 @@
  * @since 2024-05-27
  **/
 public class RocketMqSchedulerRebuildSubscriptionInterceptor extends RocketMqAbstractInterceptor {
-    private final Object lock = new Object();
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    private static final String RETYPE = "%RETRY%";
 
     @Override
     public ExecuteContext doAfter(ExecuteContext context) throws Exception {
         ConcurrentMap<String, Object> map = (ConcurrentMap<String, Object>) context.getResult();
         RebalanceImpl balance = (RebalanceImpl) context.getObject();
-        if (balance.getConsumerGroup() == null) {
+        if (balance.getConsumerGroup() == null || !isGrayTagChanged(map, balance)) {
             return context;
         }
         for (Object subscriptionData : map.values()) {
+            String topic = RocketMqReflectUtils.getTopic(subscriptionData);
+            if (topic.contains(RETYPE)) {
+                continue;
+            }
             if (RocketMqSubscriptionDataUtils
                     .isExpressionTypeInaccurate(RocketMqReflectUtils.getExpressionType(subscriptionData))) {
                 continue;
             }
```

Comment:
isGrayTagChanged has the similar code with these code. You can traverse the map only once

Reply:
fixed